### PR TITLE
Disconnect peer liang

### PR DESCRIPTION
**Pull request by lvs007**
**What does this PR do?**
If the main chain of the current node is higher than the synchronous node and the synchronization block is not found, then the peer link is disconnected.
**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: src/main/java/org/tron/common/overlay/message/ReasonCode.java [hunk 1]
```diff
@@ -82,6 +82,8 @@
 
     RESET(0x11),
 
+    SYNC_FAIL(0x12),
```
Comment (olenheim) [inline src/main/java/org/tron/common/overlay/message/ReasonCode.java]: this reason need modify proto files

File: src/main/java/org/tron/core/net/node/NodeImpl.java [hunk 2]
```diff
@@ -748,7 +749,14 @@ private void onHandleSyncBlockChainMessage(PeerConnection peer, SyncBlockChainMe
     }
 
     if (blockIds.isEmpty()) {
-      peer.setNeedSyncFromUs(false);
+      if (CollectionUtils.isNotEmpty(summaryChainIds) && summaryChainIds.get(0).getNum() < del
+          .getHeadBlockId().getNum()) {
```
Comment (olenheim) [inline src/main/java/org/tron/core/net/node/NodeImpl.java]: when peer's number id lower than our solid block number , It's really sync fail, in this case , it maybe can sync again

File: src/main/java/org/tron/common/overlay/message/ReasonCode.java (hunk 1 above)
Comment (lvs007) [reply src/main/java/org/tron/common/overlay/message/ReasonCode.java]: add to proto files

File: src/main/java/org/tron/core/net/node/NodeImpl.java (hunk 2 above)
Comment (lvs007) [reply src/main/java/org/tron/core/net/node/NodeImpl.java]: modify use LatestBlockHeaderNumber compare

**Pull request merged**